File: CPlusPlus/ValidParenthesis/ValidParenthesis.cpp
class Solution 
{
public:
    bool isValid(string s) 
    {
        string storage = "";
        
        int size = s.length();
        
            
        for(int i = 0; i < size; i++)
        {
            if (s[i] == ')')
            {
                if (storage.back() == '(')
                    storage.pop_back();
                else
                    return 0;
                
            }
            
            else if (s[i] == '}')
            {
                if (storage.back() == '{')
                    storage.pop_back();
                else
                    return 0;
            }
             
            else if (s[i] == ']')
            {
                if (storage.back() == '[')
                    storage.pop_back();
                else
                    return 0;
                
            }
            
            else
            {
                storage.push_back(s[i]);   
            }
        }
        
        if (storage.length() > 0)
            return 0;
        
        return 1;
    }
};

File: Python/Valid Parenthesis/ValidParenthesis.py


class Solution:
    def isValid(self, s: 'str') -> 'bool':
        
        validity = True
        
        storage = []
        
        for i in s:
            if i == ')':
                if not storage:
                    return False
                else:
                    if storage[-1] == '(':
                        del storage[-1]
                    else:
                        return False
            elif i == '}':
                if not storage:
                    return False
                else:
                    if storage[-1] == '{':
                        del storage[-1]
                    else:
                        return False
            elif i == ']':
                if not storage:
                    return False
                else:
                    if storage[-1] == '[':
                        del storage[-1]
                    else:
                        return False
            
            else:
                storage.append(i)
                
        if storage:
            return False
        
        return True
